Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Prerequisites for using dockerfile can be seen here https://github.com/zephyrtronium/robot/issues/76
+# docker build --build-arg CONFIG=yourConfig.toml --build-arg TWITCHSECRET=yourSecret --build-arg ROBOTKEY=yourKey -t robot .
+
+# Main image
+FROM golang:1.23-alpine AS build
+
+# Define build arguments
+ARG CONFIG \
+  TWITCHSECRET \
+  ROBOTKEY
+
+# Populate Go resources
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy repo
+COPY . .
+
+# Build Robot
+RUN --mount=type=cache,target=/root/.cache/go-build \
+  --mount=type=cache,target=/go/pkg \
+  CGO_ENABLED=0 \
+  go build -o /build/robot github.com/zephyrtronium/robot
+
+# Prepare minimised image
+FROM alpine
+
+# Set resources
+ARG CONFIG \
+  TWITCHSECRET \
+  ROBOTKEY
+ENV CONFIG=${CONFIG:-robot.toml} \
+  TWITCHSECRET=${TWITCHSECRET:-twitch_secret} \
+  ROBOTKEY=${ROBOTKEY:-robot_key}
+
+# Copy Robot binary and required resources
+COPY --from=build /build/robot /bin/robot
+COPY ${TWITCHSECRET} /
+COPY ${ROBOTKEY} /
```

Comment:
I'm not really familiar with docker secrets myself, maybe this can be marked as a follow up action for someone with more experience using them

Reply:
usually configs should not be put into the docker image since that's usually pushed to a container repository and are mounted in your `docker run` command or `docker-compose.yml` file.
You can put a default config in which the user should overwrite ofc.

As for docker secrets they are just files in the final containers located at `/run/secrets/`

Reply:
Ah, to be fair my thoughts on this was self building and self-hosting. I wasn't considering pushing the image up.

If that's the case I don't think you can make an image that starts up 

I'm guessing you'd generate a secret and point your config to the files in /run/secrets?

Assuming at this point it's better to not import any configs / keys into the image if you want to put the image on a registry and rather have a readme for how to use the image instead?

Should I remove the Config / Key / Secret?

Reply:
I can't speak for what @zephyrtronium wants, but ideally you wouldnt need to build your own image and instead can just pull it from either dockerhub or ghcr for each commit/release.

as for the secrets you'd need to have some sort of mechanism in your config to point to a secrets file instead

Reply:
It sounds logical then to remove the Dockerfile arguments, but my guess is that we'll need to expect a standardized filename for the config file then if it's mounted in as a volume

Am I correct in saying that the person mounting in the volumes for the twitch secret / robots key would need to mount it as a secret themselves, or is that something that needs to be set up in the Dockerfile?

Reply:
You are correct, that something the user needs to provide and nothing you need to handle in the dockerfile

Reply:
I'll update this tomorrow then after work. should be straightforward, mostly just removing content.